R/OntologyQueryFunctions.R

```diff
@@ -118,11 +129,39 @@ getMetaFromOnto <- function(ontology, db = RaMP()) {
 
   list_ontology <- unique(list_ontology)
 
-  allontos <- getOntologies(db = db)
-  matched_ontos <- unlist(lapply(
-    list_ontology,
-    function(x) grep(paste0("^", x, "$"), allontos$commonName)
-  ))
+  allontos <- getOntologies(db = db) %>%
+    dplyr::filter(.data$metCount <= max_ontology_size,
+                  .data$metCount > min_ontology_size)
+
+  #This if statement is meant for use with the front end.
+  #The idea is to limit the ontologies searchable in the RaMP UI to be the same ones HMDB lets the user filter by on their website.
+  #This is hard-coded as a filter here and will need to be updated manually in the current form.
+  #The default 'curate' argument should be false.
+  #Adam 10/3/2024
+  if (curate) {
+    allontos <- allontos %>% dplyr::filter(.data$commonName %in% c('Blood',
+                                                'Urine',
+                                                'Saliva',
+                                                'Cerebrospinal fluid',
+                                                'Feces',
+                                                'Sweat',
+                                                'Breast milk',
+                                                'Bile',
+                                                'Amniotic fluid',
+                                                'Exogenous',
+                                                'Endogenous',
+                                                'Food',
+                                                'Plant',
+                                                'Microbe',
+                                                'Cosmetic',
+                                                'Drug',
+                                                'Cell membrane',
+                                                'Cytoplasm',
+                                                'Nucleus',
+                                                'Mitochondria'))
+  }
+
+  matched_ontos <- list_ontology[list_ontology %in% allontos$commonName]
 
   # Only proceed if df has anything returned
   if (length(matched_ontos) > 0) {
```

Comment:
there's a bug here, but I think you didn't introduce it. I also changed some functionality around here very recently and I also failed to find or fix the bug (I don't think I introduced it either). Incidentally, you might want to pull changes and get the latest coe as well, because I changed it to pass list_ontology to the db@api directly, so callers don't have to worry about formatting the query.

below is the bug, we're filtering the list to generate matched_ontos, but then doing the query for the full list anyway list_ontology / ontologyList. so at the end we don't actually filter the list.


  # Only proceed if df has anything returned
  if (length(matched_ontos) > 0) {
    print(paste0("Found ", length(matched_ontos), " ontology term matches."))
    ontologyList <- paste0("'", paste(list_ontology, collapse = "','"), "'")
    mdf_final <- db@api$getMetabolitesForOntology(ontologyList = ontologyList)